src/main/java/org/candlepin/resource/HypervisorTypeChecker.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2009 - 2021 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.resource;
+
+import org.candlepin.model.ConsumerType;
+import org.candlepin.model.ConsumerTypeCurator;
+
+import com.google.inject.Inject;
+
+import java.util.Objects;
+
+
+public class HypervisorTypeChecker {
```

Comment:
This seems like a really complicated way of adding a `.matches`/`.isOfType`/etc. method on ConsumerType, and enum support in ConsumerTypeCurator.

Reply:
As per IRC discussion this will be done in a follow-up PR.